Decred Wallet/Features/Wallets/WalletSettingsViewController.swift

```diff
@@ -60,6 +60,36 @@ class WalletSettingsViewController: UIViewController {
         self.incomingTxAlertButton.setTitle(newSetting.localizedString, for: .normal)
     }
     
+     @IBAction func rescanBlockchain(_ sender: Any) {
+        if SyncManager.shared.isSyncing {
+            Utils.showBanner(in: self.view, type: .error, text: LocalizedStrings.errorSyncInProgress)
+        } else if !WalletLoader.shared.multiWallet.isConnectedToDecredNetwork() {
+            Utils.showBanner(in: self.view, type: .error, text: LocalizedStrings.notConnected)
+        } else if SyncManager.shared.isRescanning {
+            Utils.showBanner(in: self.view, type: .error, text: LocalizedStrings.errorRescanInProgress)
+        } else {
+            // rescan blockchain
+            self.showOkAlert(message: LocalizedStrings.rescanConfirm,
+                                 title: LocalizedStrings.rescanBlockchain,
+                                 onPressOk: self.rescanBlocks,
+                                 addCancelAction: true)
+        }
+    }
+        
+    func rescanBlocks() {
+        do {
+            try WalletLoader.shared.multiWallet.rescanBlocks(self.wallet.id_)
+            DispatchQueue.main.async {
+                Utils.showBanner(in: self.view, type: .success, text: LocalizedStrings.rescanProgressNotification)
+            }
+        } catch let error {
+            var errorMessage = error.localizedDescription
+            if errorMessage == DcrlibwalletErrInvalid {
+                errorMessage = "scan started already"
+            }
```

Comment:
Well, what do you do with this error message? Probably log AND display to user.

Reply:
i will just log it. the error was handled and displayed to the user before this function call.